nixos/modules/services/misc/gitea.nix

```diff
@@ -0,0 +1,405 @@
+    PASSWD = ${cfg.mail.password}
+
+    [other]
+    SHOW_FOOTER_BRANDING = ${toConfigBool cfg.footer.branding}
+    SHOW_FOOTER_VERSION = ${toConfigBool cfg.footer.version}
+    SHOW_FOOTER_TEMPLATE_LOAD_TIME = ${toConfigBool cfg.footer.loadtime}
+
+    [log]
+    LEVEL = Warn
+
+    ${cfg.extraConfig}
+  '';
+in
+
+{
+  options = {
+    services.gitea = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable Go Git Service.";
+      };
+
+      useWizard = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Use Gitea installation wizard instead of creating a 
+          configuration file.
+          The first registered user will be administrator.
+        '';
+      };
+
+      appName = mkOption {
+        type = types.str;
+        default = "Gitea: Git with a cup of tea";
+        description = "Application name.";
+      };
+
+      repository = {
+        root = mkOption {
+          type = types.str;
+          default = "${cfg.stateDir}/repositories";
+          description = "Path to the git repositories.";
+        };
+        forcePrivate = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Force every new repository to be private.";
+        };
+      };
+
+      server = {
+        domain = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = "Domain name of your server.";
+        };
+
+        rootUrl = mkOption {
+          type = types.str;
+          default = "http://localhost:3000/";
+          description = "Full public URL of Gitea server.";
+        };
+
+        httpAddress = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          description = "HTTP listen address.";
+        };
+
+        httpPort = mkOption {
+          type = types.int;
+          default = 3000;
+          description = "HTTP listen port.";
+        };
+      };
+
+      database = {
+        type = mkOption {
+          type = types.enum [ "sqlite3" "mysql" "postgres" ];
+          example = "mysql";
+          default = "sqlite3";
+          description = "Database engine to use.";
+        };
+
+        host = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = "Database host address.";
+        };
+
+        port = mkOption {
+          type = types.int;
+          default = 3306;
+          description = "Database host port.";
+        };
+
+        name = mkOption {
+          type = types.str;
+          default = "gitea";
+          description = "Database name.";
+        };
+
+        user = mkOption {
+          type = types.str;
+          default = "gitea";
+          description = "Database user.";
+        };
+
+        password = mkOption {
+          type = types.str;
+          default = "";
+          description = "Database password.";
```

Comment:
There should be an option to store this in a external file. Explanation: https://github.com/NixOS/nixpkgs/issues/24288